### ISSUE TITLE
Use JsonConverter if they are available

### ISSUE THREAD

**Issue by dvorak-jan** [elliotritchie/NES#34]
Hello,
I have a suggestion: use JsonConverter collection if it is available in JsonConvert.DefaultSettings().Converters or (optionally) let user pass converters or whole settings in wire-up method UseJsonSerialization().

It would require just small non-breaking change here https://github.com/elliotritchie/NES/blob/master/src/NES.NEventStore/JsonSerializer.cs#L21

something along the lines:

```
public JsonSerializer(Func<IEventMapper> eventMapperFunc, Func<IEventFactory> eventFactoryFunc)
{
    _settings = GetJsonSerializerSettings();
    _eventMapperFunc = eventMapperFunc;
    _eventFactoryFunc = eventFactoryFunc;
}

private static JsonSerializerSettings GetJsonSerializerSettings()
{
    var jss = JsonConvert.DefaultSettings != null 
        ? JsonConvert.DefaultSettings.Invoke() 
        : new JsonSerializerSettings();

    jss.TypeNameAssemblyFormat = FormatterAssemblyStyle.Simple;
    jss.TypeNameHandling = TypeNameHandling.Auto;
    jss.DefaultValueHandling = DefaultValueHandling.Ignore;
    jss.NullValueHandling = NullValueHandling.Ignore;

    return jss;
}

...
```

**Comment by marinkobabic** [comment]
I think it makes more sense to pass the settings optional into the UserJsonSerializer. It's more explicit. Do you need it immediately?

**Comment by marinkobabic** [comment]
It's implemented in the actual nuget package 5.0.7. Settings can be passed as parameter
